gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageImpl.java

```diff
@@ -2129,7 +2129,18 @@ private StorageObject getObject(StorageResourceId resourceId) throws IOException
             // https://cloud.google.com/storage/docs/json_api/v1/objects#resource-representations
             .setFields(OBJECT_FIELDS);
     try {
-      return getObject.execute();
+      long startTimeOfMetadataRead = System.currentTimeMillis();
+      StorageObject object = getObject.execute();
+      long requestDelay = System.currentTimeMillis() - startTimeOfMetadataRead;
+      logger.atFine().log(
+          "GoogleCloudStorageImpl:getMetadata complete context:%d,time:%d,resource:%s,requestId:%s",
+          Thread.currentThread().getId(),
+          requestDelay,
+          resourceId,
+          requestDelay > 1000
+              ? getObject.getLastResponseHeaders().getFirstHeaderStringValue("x-guploader-uploadid")
```

Comment:
Should we log this upload-id all the time? If too much log is the issue, then anyway we are logging the remaining string.

Reply:
+1 Let's move this to trace logs (FINEST) and log every time